client/src/components/auth-modal-form.tsx

```diff
@@ -23,6 +23,31 @@ import { SimpleLoader } from "../design-system/components/simple-loader";
 import { useIsWaitingForTooLong } from "../hooks/use-is-waiting-for-too-long";
 import { User } from "@/lib/storage/domain";
 import { getUserService } from "@/domains/user/user-service";
+import { Eye, EyeOff } from "lucide-react";
+
+// Reusable component for a password field with visibility toggle
+const PasswordInput = ({ ...props }: React.ComponentProps<typeof Input>) => {
+  const [showPassword, setShowPassword] = useState(false);
+
+  return (
+    <div className="relative">
+      <Input type={showPassword ? "text" : "password"} {...props} />
+      <Button
+        type="button"
+        variant="ghost"
+        size="sm"
+        className="absolute top-0 right-0 h-full px-3 py-2 hover:bg-transparent"
+        onClick={() => setShowPassword(!showPassword)}
```

Comment:
je pense que je préfère la syntaxe avec callback ici, histoire d'être sûr d'avoir la version à jour du state. Dans les faits les deux marchent, mais je pense que `(prev) => !prev` est plus standard

```suggestion
        onClick={() => setShowPassword((prev) => !prev)}
```